sympy/geometry/point.py

```diff
@@ -516,8 +516,12 @@ def transform(self, matrix):
         geometry.entity.scale
         geometry.entity.translate
         """
-        x, y = self.args
-        return Point(*(Matrix(1, 3, [x, y, 1])*matrix).tolist()[0][:2])
+        if isinstance(matrix, Matrix) and matrix.shape == (3, 3):
+            x, y = self.args
+            return Point(*(Matrix(1, 3, [x, y, 1])*matrix).tolist()[0][:2])
+        else:
+            raise ValueError("The argument to the transform function must be " \
+            + "a 3x3 matrix")
```

Comment:
It's better to use this pattern:

```
if not isinstance(...) or not matrix.shape == ...:
  raise ValueError(...)
# original code goes here
```

This unnests code, so it's easier to read. (I.e. the productive code (original lines 519 f.) isn't buried deep down in a nested `if`.)
Also, if somebody wants to add additional checks, or split the check in two, adding an additional check doesn't mean you have to touch the productive code.
It also doesn't touch the original code, that makes it easier to see who authored it in `git annotate`.